tests/js/plugin/enhancedanchoring-test.coffee

```diff
@@ -634,3 +657,64 @@ describe 'Annotator.Plugin.EnhancedAnchoring', ->
             anchor.realize()
 
             assert anchor.fullyRealized
+
+    describe 'when scrolling to a virtual anchor', ->
+
+      it 'scrolls right next to the wanted page (to get it rendered)', ->
+        am = createAnchoringManagerAndLazyDocument()
+        ann = createTestAnnotationForPages "a1", [10]
+        anchor = am.createAnchor(ann, ann.target[0]).result
+        am.document.currentIndex = 5  # We start from page 5
+        am.document.setPageIndex = sinon.spy (index) ->
+          am.document.currentIndex = index
+          if index is 9
+            renderPage am.document, 9
+            renderPage am.document, 10
+
+        # Now we trigger the actual action
+        anchor.scrollIntoView().then ->
+          assert.calledWith am.document.setPageIndex, 9
+
+      it 'gets the wanted page rendered', ->
+        am = createAnchoringManagerAndLazyDocument()
+        ann = createTestAnnotationForPages "a1", [10]
+        anchor = am.createAnchor(ann, ann.target[0]).result
+        am.document.currentIndex = 5  # We start from page 5
+        am.document.setPageIndex = sinon.spy (index) ->
+          am.document.currentIndex = index
+          if index is 9
+            renderPage am.document, 9
+            renderPage am.document, 10
+
+        # Now we trigger the actual action
+        anchor.scrollIntoView().then ->
+          assert am.document.isPageMapped 10
+
+      it 'scrolls to the wanted page, eventually', ->
+        am = createAnchoringManagerAndLazyDocument()
+        ann = createTestAnnotationForPages "a1", [10]
+        anchor = am.createAnchor(ann, ann.target[0]).result
+        am.document.currentIndex = 5  # We start from page 5
+        am.document.setPageIndex = sinon.spy (index) ->
+          am.document.currentIndex = index
+          if index is 9
+            renderPage am.document, 9
+            renderPage am.document, 10
```

Comment:
> This looks like it's testing the wiring between the `setPageIndex` and the `renderPage` helper. These tests should only be validating the behaviour of the `scrollIntoView` method and that they call appropriate methods on dependencies. The behaviour of those dependencies should be tested elsewhere.

To also give a shorter, more concrete answer:
 * The `Anchoring Manager` (which is the subject of this test) calls the `docAccess`'s `setPageIndex()` method as the first phase of the scrolling
 * To proceed to testing the second phase of the scrolling, the `docAccess` must get the `platform` render the page that we need.
 * That's why we are mocking the `setPageIndex()` method, and use the (fake) `renderPage()` method. What we are doing here is the bare minimum to short-circuit the system, so that the next function of the `Anchoring Manager` (which is rendering the highlight on the newly rendered page, and the scrolling to it) can be exercised.